apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/OffsetLinkBuilder.java

```diff
@@ -48,14 +48,27 @@ public final class OffsetLinkBuilder implements LinkBuilder {
    */
   public OffsetLinkBuilder(RequestDetails requestDetails, String resource) {
     this.pageSize = parseIntegerParameters(requestDetails, Constants.PARAM_COUNT);
-    startIndex = parseIntegerParameters(requestDetails, "startIndex");
+    this.startIndex = parseIntegerParameters(requestDetails, "startIndex");
+    this.serverBase = requestDetails.getServerBaseForRequest();
+    this.resource = resource;
+    this.requestDetails = requestDetails;
+    validate();
```

Comment:
This is some cleanup to make this class do all its validations upfront, like PatientLinkBuilder does. Made more sense to me than waiting until we call the parameter to bother checking if its valid. (Not technically related to the fix, but I was in this code poking around for answers)

Reply:
makes sense!

Reply:
Where would this `IllegalRequestException` be handled if it's thrown in the constructor?

Reply:
It gets caught by HAPI-FHIR and turned into a 400. It's HAPI-FHIR's exception, and the only way to get it to throw 400s which is kinda gross since it ties us to the implementation, but every non-HAPI exception creates a 500 in their internal code.